### PR TITLE
LeNet + CIFAR-10 example

### PR DESCRIPTION
**Pull request by stites**
Description should be updated later:
- adds `hasktorch-zoo` -- this will, hopefully, kick off the "call for collaboration."
- `hasktorch-zoo` includes a really dumb dataloader. Dataloading is sort of the bottleneck here.
- cabal2nix all the packages (even though this doesn't really work until we can build ATen with nix) (re: #103 , @tstat )

I think, when this is done, we are ready for hackage and make a subreddit announcement!

TODO:
- [ ] Fix exploding loss function
- [ ] stack minibatches into tensors.
- [ ] Shuffle training and testing data

### CODE REVIEW

**Comment by austinvhuang** [comment]
Going ahead and merging so I'll work off this, note to add missing gd dependencies to fix ci.
